lib/json5/lexer.ml

```diff
@@ -9,34 +9,40 @@ type token =
   | CLOSE_BRACKET
   | COLON
   | COMMA
-  | COMMENT of string
   | TRUE
   | FALSE
   | NULL
   | FLOAT of string
   | INT_OR_FLOAT of string
   | INT of string
   | STRING of string
-  | IDENTIFIER_NAME of string
+  | EOF
 
 let pp_token ppf = function
-  | OPEN_PAREN -> Format.fprintf ppf "'('"
-  | CLOSE_PAREN -> Format.fprintf ppf "')'"
-  | OPEN_BRACE -> Format.fprintf ppf "'{'"
-  | CLOSE_BRACE -> Format.fprintf ppf "'}'"
-  | OPEN_BRACKET -> Format.fprintf ppf "'['"
-  | CLOSE_BRACKET -> Format.fprintf ppf "']'"
-  | COLON -> Format.fprintf ppf "':'"
-  | COMMA -> Format.fprintf ppf "','"
-  | COMMENT s -> Format.fprintf ppf "COMMENT '%s'" s
-  | TRUE -> Format.fprintf ppf "'true'"
-  | FALSE -> Format.fprintf ppf "'false'"
-  | NULL -> Format.fprintf ppf "'null'"
-  | FLOAT s -> Format.fprintf ppf "FLOAT '%s'" s
-  | INT_OR_FLOAT s -> Format.fprintf ppf "INT_OR_FLOAT '%s'" s
-  | INT s -> Format.fprintf ppf "INT '%s'" s
-  | STRING s -> Format.fprintf ppf "STRING '%s'" s
-  | IDENTIFIER_NAME s -> Format.fprintf ppf "IDENTIFIER_NAME '%s'" s
+  | OPEN_PAREN -> Format.pp_print_string ppf "("
+  | CLOSE_PAREN -> Format.pp_print_string ppf ")"
+  | OPEN_BRACE -> Format.pp_print_string ppf "{"
+  | CLOSE_BRACE -> Format.pp_print_string ppf "}"
+  | OPEN_BRACKET -> Format.pp_print_string ppf "["
+  | CLOSE_BRACKET -> Format.pp_print_string ppf "]"
+  | COLON -> Format.pp_print_string ppf ":"
+  | COMMA -> Format.pp_print_string ppf ","
+  | TRUE -> Format.pp_print_string ppf "true"
+  | FALSE -> Format.pp_print_string ppf "false"
+  | NULL -> Format.pp_print_string ppf "null"
+  | FLOAT s | INT_OR_FLOAT s | INT s -> Format.pp_print_string ppf s
+  | STRING s -> Format.fprintf ppf "%S" s
+  | EOF -> Format.pp_print_string ppf "eof"
+
+let custom_error lexbuf =
+  let { Lexing.pos_fname = file; pos_lnum = line; _ }, _ =
+    Sedlexing.lexing_positions lexbuf
+  in
+  let file_line =
```

Comment:
This logic is duplicated in 3 places, possibly reasonable to join them to be consistent.

Reply:
What do you suggest instead?
I can share the two ones that are in the yojson5 module.

Reply:
I've put them in a shared module.